ext/sass/Rakefile

```diff
@@ -302,46 +418,140 @@ module FileUtils
   end
 end
 
-# The {SassConfig} module.
-module SassConfig
-  module Platform
-    OS = case RbConfig::CONFIG['host_os'].downcase
-         when /darwin/
-           'darwin'
-         when /linux-android/
-           'linux-android'
-         when /linux-musl/
-           'linux-musl'
-         when /linux-uclibc/
-           'linux-uclibc'
-         when /linux/
-           'linux'
-         when *Gem::WIN_PATTERNS
-           'windows'
-         else
-           RbConfig::CONFIG['host_os'].downcase
-         end
+# The {Platform} module.
+module Platform
+  # @see https://docs.freebsd.org/en/articles/linux-emulation/
+  # @see https://docs.freebsd.org/en/books/handbook/linuxemu/
+  module Linuxulator
+    module_function
 
-    CPU = case RbConfig::CONFIG['host_cpu'].downcase
-          when /amd64|x86_64|x64/
-            'x86_64'
-          when /i\d86|x86|i86pc/
-            'i386'
-          when /arm64|aarch64/
-            'aarch64'
-          when /arm/
-            'arm'
-          when /ppc64le|powerpc64le/
-            'ppc64le'
-          else
-            RbConfig::CONFIG['host_cpu']
-          end
+    def enabled?
+      return false unless RbConfig::CONFIG['host_os'].include?('freebsd')
+
+      return true if defined?(Platform::OS) && Platform::OS.include?('linux')
+
+      begin
+        Rake::Task['true'].invoke unless File.exist?('true')
+      rescue NotImplementedError
+        # do nothing
+      end
 
-    ARCH = "#{CPU}-#{OS}".freeze
+      system('./true') == true
+    end
+
+    def host_os(root = compat_linux_emul_path)
+      return 'linux-none' unless File.exist?(File.absolute_path('proc/self/exe', root))
```

Comment:
Make sure linprocfs is mounted or just pretend we cannot run dynamic executable, as dart may crash with:
```
../../runtime/vm/os_thread.cc: 49: error: Failed to retrieve stack bounds
```